packages/segmented-control/src/SegmentedControlOption.styles.ts

```diff
@@ -0,0 +1,198 @@
+  theme,
+  size = 'default',
+  baseFontSize = 14,
+}: {
+  theme: Theme;
+  size: Size;
+  baseFontSize: 14 | 16;
+}) =>
+  cx(
+    optionThemeStyle[theme],
+    optionSizeStyle[size],
+    css`
+      position: relative;
+      display: flex;
+      width: 100%;
+      align-items: center;
+      justify-content: center;
+      z-index: 3;
+
+      &:first-child,
+      &[data-lg-checked='true'],
+      &[data-lg-checked='true'] + [data-lg-checked='false'],
+      &:focus-within + :not(:focus-within) {
+        --divider-background-color: transparent;
+      }
+
+      /* 
+      * Adds the divider line to unselected segments 
+      */
+      &:before {
+        --divider-width: 1px;
+        content: '';
+        position: absolute;
+        height: var(--divider-height);
+        width: var(--divider-width);
+        left: calc(0px - (var(--segment-gap) + var(--divider-width)) / 2);
+        top: calc(
+          (
+              var(--line-height) + var(--padding-block) * 2 -
+                var(--divider-height)
+            ) / 2
+        );
+        transition: background-color 150ms ease-in-out;
+        background-color: var(--divider-background-color);
+      }
+    `,
+    {
+      // Update font size according to baseFontSize
+      [css`
+        --font-size: 16px;
+      `]: size === 'default' && baseFontSize === 16,
+    },
+  );
+
+export const boxStyle = css`
+  width: 100%;
+  height: 100%;
+  text-decoration: none;
+`;
+
+export const buttonStyle = css`
+  font-family: ${fontFamilies.default};
+  display: inline-flex;
+  position: relative;
+  width: 100%;
+  height: 100%;
+  align-items: center;
+  justify-content: center;
+  padding: var(--padding-block) var(--padding-inline);
+  background-color: var(--base-background-color);
+  border-radius: var(--indicator-radius);
+  text-align: center;
+  font-size: var(--font-size);
+  line-height: var(--line-height);
+  text-transform: var(--text-transform, none);
+  font-weight: var(--font-weight);
+  color: var(--base-text-color);
+  box-shadow: 0px 1px 2px var(--base-shadow-color);
+  cursor: pointer;
+  transition: 150ms ease-in-out;
+  transition-property: color, box-shadow;
+  text-decoration: none;
+  outline: none;
+  border: none;
+
+  &:hover {
+    color: var(--hover-text-color);
+  }
+
+  &[aria-selected='true'] {
+    color: var(--active-text-color);
+  }
+
+  &:disabled {
+    color: var(--disabled-text-color);
+    cursor: not-allowed;
+  }
+`;
+
+export const buttonFocusStyle: Record<Mode, string> = {
+  [Theme.Light]: css`
+    &:focus {
+      box-shadow: ${focusRing.light.default};
+    }
+  `,
+  [Theme.Dark]: css`
+    &:focus {
+      box-shadow: ${focusRing.dark.default};
+    }
+  `,
+};
+
+export const labelStyle = css`
+  min-height: var(--line-height);
+  gap: calc(var(--font-size) / 2);
+  display: flex;
+  align-items: center;
+  min-width: 0;
+
+  span {
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
```

Comment:
Also, don't love that it _has_ to be a `span`

Reply:
I can make it a `div`

Reply:
Semantically it makes sense, but the fact that a user would _have to_ use a `span` and nothing else doesn't feel right

Reply:
oh, that makes sense! I guess it doesn't matter now since we're adding in the component.